### ISSUE TITLE
Make nospectre_v2 work on 64-bit book3s

### ISSUE THREAD

**Issue by mpe** [linuxppc/issues#236]
Currently we only support the `nospectre_v2` command line option with book3e (FSL).

That's because the original spectre v2 mitigations were not under kernel control, we were simply told what the mitigation was, so we couldn't support turning those off at runtime.

However on some newer CPUs the spectre v2 mitigation is to flush the count cache at context switch, which is implemented by the kernel. So on those CPUs we could support `nospectre_v2`.

The logic should probably go in `setup_count_cache_flush()`. We would want to do something like:

```
if (no_spectrev2) {
	if (security_ftr_enabled(SEC_FTR_BCCTRL_SERIALISED) ||
	    security_ftr_enabled(SEC_FTR_COUNT_CACHE_DISABLED))
		pr_warn("Spectre v2 mitigations not under software control, can't disable")

	return;
}

toggle_count_cache_flush(true);
```

**Comment by cmr-informatik** [comment]
I can take a peek at this if there are no takers.

**Comment by mpe** [comment]
Thanks.

**Comment by ajdlinux** [comment]
current patch: https://patchwork.ozlabs.org/patch/1104583/

**Comment by cmr-informatik** [comment]
the patch in the link posted by @ajdlinux compiles and works (tested w/ QEMU and ppc64le_defconfig) on linuxppc/next now

**Comment by mpe** [comment]
Merged as torvalds/linux@d8f0e0b073e1, thanks @cmr-informatik :ice_cream:
